### PR TITLE
Lab: add badge checking bots and spawns

### PR DESCRIPTION
**Pull request by curstwist**
#### Summary
none

#### Purpose of change
add more security defenses to the  new lab

#### Describe the solution

- Add a set of 6 drones that check your security access eligibility when they encounter you.  These use a combination of spell mutation & EOC to do the badge scan, checks, and subsequent behaviors.
- Spawn them in the lab in their appropriate zones 
- Add lights to the OMTs that got the spawns.
- Rework the 6th level security corridor to be more exciting and different then the 5th level.

#### Describe alternatives you've considered
I'd like to do more work on these bots, but would like player feedback before I invest too much time.

I've given these drones their own file in monsters since they are pretty complex.  I may need to break them up more in the future if I continue to expand on them.

#### Testing

Debug testing in game.

#### Additional context
 I may make some minor adjustments but this lab is almost done except for Venera3's content (which is essential to release) and entrance card spawns/start locations.

<details>
  <summary>Spoiler warning</summary>
  
The bots are meant to offer a way to trigger mapgen_update in a generally defined area and be more reactive to player decisions.  They will keep scanning you if you don't remove them.  If you keep your badge on, they should remain neutral and some will give you helpful snippets (the maintenance area ones).  If you drop your badge, they will react the next time they scan you.  Their response may take a few scans or a single scan.  They unleash various robot mobs that I've stashed around the map.  I've made them fairly strong so they player will need to invest time to disable them or come prepared.

Not all mapgen updates for bot release are tied to these badge bots.  Some use an RNG terrain trigger (standard player steps on tile) and some are EOC tied to a particular OMT.  I think playtesting will show us which are effective and which should be altered to another type.

If a player collects and keeps all the badges, they will have a much safer lab experience then if they opt to break in and ignore the security measures.

Note: badges aren't consumed per use but are limited to the lab they are spawned in (except entrance passes which will be universal, one time cards).  Green Researcher and Black Security Badges aren't spawning yet and will need debug spawning if you want to test them.

</details>

### CODE REVIEW

File: data/json/monsters/lab_security_drones.json [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+    "description": "A silent mutation to work with lab security drones and badges.",
+    "purifiable": false,
+    "player_display": false
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "bot_green_maintenance_badge_check",
+    "recurrence_min": 10,
+    "recurrence_max": 60,
+    "condition": { "and": [ { "not": { "u_has_item": "id_science_maintenance_green" } }, { "u_has_trait": "GREEN_MAINTENANCE_ALARM" } ] },
```
Comment (actual-nh) [inline data/json/monsters/lab_security_drones.json:33]: So this only happens if you already have the GREEN_MAINTENANCE_ALARM trait? It does not make much sense to add the trait if you already have it.

**Review by actual-nh — COMMENTED** [review]
Nice! Just a few things.

File: data/json/items/generic.json [hunk 2]
```diff
@@ -868,7 +868,6 @@
     "material": [ "steel", "iron" ],
     "weight": "22680 g",
     "volume": "6250 ml",
-    "longest_side": "1 meter",
```
[the remark this answers is not in the document]
Comment (curstwist) [reply data/json/items/generic.json:871]: mistakenly reverted this addition

File: data/json/monsters/lab_security_drones.json (hunk 1 above)
Comment (curstwist) [reply data/json/monsters/lab_security_drones.json:33]: pulled these out and did some testing.  They appear to be redundant and not effecting the EOC chain so I've removed them.

File: data/json/items/generic.json [hunk 3]
```diff
@@ -868,7 +868,6 @@
     "material": [ "steel", "iron" ],
     "weight": "22680 g",
     "volume": "6250 ml",
```
Comment (kevingranade) [inline data/json/items/generic.json:870]:
```suggestion
    "volume": "6250 ml",
    "longest_side": "1 meter",
```

**Pull request merged**